libs/src/sanityChecks/rolloverNodes.js

```diff
@@ -1,69 +1,88 @@
 
 const THREE_SECONDS = 3000
 const MAX_TRIES = 3
 
-/** Check if the user's primary creator node is healthy */
-const checkPrimaryHealthy = async (libs, primary, tries) => {
-  const healthy = await Utils.isHealthy(primary)
+const nodeIsHealthy = async (endpoint, tries) => {
+  const healthy = await Utils.isHealthy(endpoint)
   if (healthy) return healthy
   else {
-    if (tries === 0) {
-      return false
-    }
+    if (tries === 0) return false
     await Utils.wait(THREE_SECONDS)
-    return checkPrimaryHealthy(libs, primary, tries - 1)
+    return nodeIsHealthy(endpoint, tries - 1)
   }
 }
 
-/** Gets new endpoints from a user's secondaries */
-const getNewPrimary = async (libs, secondaries) => {
+const getUpToDateSecondary = async (libs, secondaries) => {
   for (const secondary of secondaries) {
     const { isBehind } = await libs.creatorNode.getSyncStatus(secondary)
     if (!isBehind) {
       return secondary
     }
   }
-  throw new Error(`Could not find valid secondaries for user ${secondaries}`)
+  return null
 }
 
+/** If selected primary or secondaries unhealthy, select new one and update user state. */
 const rolloverNodes = async (libs, creatorNodeWhitelist) => {
   const user = libs.userStateManager.getCurrentUser()
 
   if (!user || !user.is_creator) return
 
   const primary = CreatorNode.getPrimary(user.creator_node_endpoint)
-  const healthy = await checkPrimaryHealthy(libs, primary, MAX_TRIES)
-  if (healthy) return
-
   const secondaries = CreatorNode.getSecondaries(user.creator_node_endpoint)
 
   try {
-    // Get a new primary
-    const newPrimary = await getNewPrimary(libs, secondaries)
-    const index = secondaries.indexOf(newPrimary)
-    // Get new secondaries and backfill up to 2
+    let newPrimary = primary
     let newSecondaries = [...secondaries]
-    newSecondaries.splice(index, 1)
-    const autoselect = await libs.ServiceProvider.autoSelectCreatorNodes(
+
+    // Select new primary from secondaries if necessary, and remove that node from secondaries list.
+    if (!await nodeIsHealthy(primary, MAX_TRIES)) {
+      const upToDateSecondary = await getUpToDateSecondary(libs, secondaries)
+      if (!upToDateSecondary) {
+        throw new Error('Cannot select new primary - no valid secondaries found for user.')
+      }
+      newPrimary = upToDateSecondary
+      newSecondaries.splice(secondaries.indexOf(upToDateSecondary), 1)
+    }
+
+    // Filter out unhealthy secondaries.
+    // TODO - see if possible to replace with filter with await isHealthy check
+    const unhealthySecondaries = []
+    for (const secondary of newSecondaries) {
+      if (!await nodeIsHealthy(secondary, MAX_TRIES)) {
+        unhealthySecondaries.push(secondary)
+      }
+    }
+    newSecondaries = newSecondaries.filter(s => !unhealthySecondaries.includes(s))
+
+    if (newSecondaries.length >= 2 && _.isEqual(primary, newPrimary) && _.isEqual(secondaries, newSecondaries)) {
```

Comment:
I'm not sure if i totally understand this check--if there's no difference in the stuff we selected, just return? when does this happen? when stuff is momentarily down, but then comes back while this fn is running? we've already spliced out a secondary, so when would newSecondaries.length >= 2?